Mathlib/LinearAlgebra/Matrix/SesquilinearForm.lean

```diff
@@ -80,196 +89,205 @@ section AuxToMatrix
       f := by
   refine ext_basis (Pi.basisFun R₁ n) (Pi.basisFun R₂ m) fun i j => ?_
   simp_rw [Pi.basisFun_apply, Matrix.toLinearMap₂'Aux_stdBasis, LinearMap.toMatrix₂Aux_apply]
 #align linear_map.to_linear_map₂'_aux_to_matrix₂_aux LinearMap.toLinearMap₂'Aux_toMatrix₂Aux
 
-theorem Matrix.toMatrix₂Aux_toLinearMap₂'Aux (f : Matrix n m R) :
-    LinearMap.toMatrix₂Aux (fun i => LinearMap.stdBasis R₁ (fun _ => R₁) i 1)
+theorem Matrix.toMatrix₂Aux_toLinearMap₂'Aux (f : Matrix n m N₂) :
+    LinearMap.toMatrix₂Aux (R := R) (fun i => LinearMap.stdBasis R₁ (fun _ => R₁) i 1)
         (fun j => LinearMap.stdBasis R₂ (fun _ => R₂) j 1) (f.toLinearMap₂'Aux σ₁ σ₂) =
       f := by
   ext i j
   simp_rw [LinearMap.toMatrix₂Aux_apply, Matrix.toLinearMap₂'Aux_stdBasis]
 #align matrix.to_matrix₂_aux_to_linear_map₂'_aux Matrix.toMatrix₂Aux_toLinearMap₂'Aux
 
-end CommRing
+end CommSemiring
 
 end AuxToMatrix
 
 section ToMatrix'
 
-/-! ### Bilinear forms over `n → R`
+/-! ### Bilinear maps over `n → R`
 
-This section deals with the conversion between matrices and sesquilinear forms on `n → R`.
+This section deals with the conversion between matrices and sesquilinear maps on `n → R`.
 -/
 
-
-variable [CommSemiring R] [Semiring R₁] [Semiring R₂]
+variable [CommSemiring R] [AddCommMonoid N₂] [Module R N₂] [Semiring R₁] [Semiring R₂]
+  [SMulCommClass R R N₂] [Semiring S₁] [Semiring S₂] [Module S₁ N₂] [Module S₂ N₂]
+  [SMulCommClass S₁ R N₂] [SMulCommClass S₂ R N₂] [SMulCommClass S₂ S₁ N₂]
+variable {σ₁ : R₁ →+* S₁} {σ₂ : R₂ →+* S₂}
 variable [Fintype n] [Fintype m]
 variable [DecidableEq n] [DecidableEq m]
-variable {σ₁ : R₁ →+* R} {σ₂ : R₂ →+* R}
 
-/-- The linear equivalence between sesquilinear forms and `n × m` matrices -/
-def LinearMap.toMatrixₛₗ₂' : ((n → R₁) →ₛₗ[σ₁] (m → R₂) →ₛₗ[σ₂] R) ≃ₗ[R] Matrix n m R :=
-  {
-    LinearMap.toMatrix₂Aux (fun i => stdBasis R₁ (fun _ => R₁) i 1) fun j =>
+/-- The linear equivalence between sesquilinear maps and `n × m` matrices -/
+def LinearMap.toMatrixₛₗ₂' : ((n → R₁) →ₛₗ[σ₁] (m → R₂) →ₛₗ[σ₂] N₂) ≃ₗ[R] Matrix n m N₂ :=
+  { LinearMap.toMatrix₂Aux (R := R) (fun i => stdBasis R₁ (fun _ => R₁) i 1) fun j =>
       stdBasis R₂ (fun _ => R₂) j
         1 with
-    toFun := LinearMap.toMatrix₂Aux _ _
+    toFun := LinearMap.toMatrix₂Aux (R := R) _ _
     invFun := Matrix.toLinearMap₂'Aux σ₁ σ₂
-    left_inv := LinearMap.toLinearMap₂'Aux_toMatrix₂Aux
-    right_inv := Matrix.toMatrix₂Aux_toLinearMap₂'Aux }
+    left_inv := LinearMap.toLinearMap₂'Aux_toMatrix₂Aux (R := R)
+    right_inv := Matrix.toMatrix₂Aux_toLinearMap₂'Aux (R := R) }
 #align linear_map.to_matrixₛₗ₂' LinearMap.toMatrixₛₗ₂'
 
-/-- The linear equivalence between bilinear forms and `n × m` matrices -/
-def LinearMap.toMatrix₂' : ((n → R) →ₗ[R] (m → R) →ₗ[R] R) ≃ₗ[R] Matrix n m R :=
+/-- The linear equivalence between bilinear maps and `n × m` matrices -/
+def LinearMap.toMatrix₂' : ((n → S₁) →ₗ[S₁] (m → S₂) →ₗ[S₂] N₂) ≃ₗ[R] Matrix n m N₂ :=
```

Comment:
It seems that because of the changes, then `R` argument is almost never inferred, which is why you need `(R := R)` all over the place. I would just make this argument explicit to this function. Then everywhere, just use `R` instead of `(R := R)`. This may improve the output of the benchmark too.

Note that this behavior is not unexpected: in the previous iteration, the `R` argument could be inferred frequently from the first argument to the linear map when it was applied to some element, but that is no longer possible because `R` no longer appears in the type of that argument (or in the output argument).